app/src/main/java/tech/relaycorp/courier/data/model/MessageAddress.kt

```diff
@@ -33,5 +38,31 @@ sealed class MessageAddress {
     }
 }
 
-data class PublicMessageAddress(val publicValue: String) : MessageAddress()
+data class PublicMessageAddress(val publicValue: String) : MessageAddress() {
+    @Suppress("BlockingMethodInNonBlockingContext") // Needed for URL()
+    @Throws(PublicAddressResolutionException::class)
+    suspend fun resolve(doHClient: DoHClient): String {
```

Comment:
I wouldn't have place this inside the data model, but instead on a separate resolver module (if this isn't something that could be inside the DoHClient itself). But I know we've done this pattern on the relaynet-jvm library too, where RAMFMessage objects have methods to validate and resolve things inside. So I'm fine with letting it be here.